### PR TITLE
Add stage-release workflow for 2.6

### PR DESCRIPTION
**Pull request by aiyengar2**
Blocked on https://github.com/rancher/charts/pull/1136.

Related Issue: https://github.com/rancher/rancher/issues/32146

### CODE REVIEW

File: .github/workflows/push-2.6.yaml [hunk 1]
```diff
@@ -28,3 +28,12 @@ jobs:
...
       run: git push origin dev-v2.6
+
+    - name: Stage released/ directory
+      run: FROM_WORKFLOW=1 RELEASE_REPO_URL=https://github.com/rancher/charts.git RELEASE_BRANCH=release-v2.6 make stage-release
+
+    - name: Add released/ to branch
+      run: git add . && git -c user.name="actions" -c user.email="actions@github.com" commit -m "Regenerate released directory" -m "$(git log -b dev-v2.5-source --oneline -n1 --pretty=format:'%B')" || true
+
+    - name: Push assets
+      run: git push origin dev-v2.6
```
Comment (cbron) [inline .github/workflows/push-2.6.yaml:39]: Whats point of adding all this to 2.6 if we are going to make them fully separate branches ?

Comment (aiyengar2) [reply .github/workflows/push-2.6.yaml:39]: At the current moment, they aren't separate, but when we separate them, we can drop this Github Workflow entirely

**Review by thedadams — APPROVED** [review]

File: .github/workflows/push-2.6.yaml (hunk 1 above)
Comment (aiyengar2) [reply .github/workflows/push-2.6.yaml:39]: We could repurpose this PR to drop the 2.6 workflows entirely in favor of manual forward-ports.

Comment (cbron) [reply .github/workflows/push-2.6.yaml:39]: Talked offline, we should keep this working for now until we confirm its being repurposed.

**Review by cbron — APPROVED** [review]

**Review by brendarearden — APPROVED** [review]

**Pull request merged**